x-pack/solutions/security/plugins/security_solution/public/flyout/ai_for_soc/components/take_action_button.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { EuiButton, EuiContextMenu, EuiPopover } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useAIForSOCDetailsContext } from '../context';
+import { useAddToCaseActions } from '../../../detections/components/alerts_table/timeline_actions/use_add_to_case_actions';
+import { useAlertTagsActions } from '../../../detections/components/alerts_table/timeline_actions/use_alert_tags_actions';
+
+export const TAKE_ACTION_BUTTON_TEST_ID = 'alert-summary-flyout-take-action';
+
+export const TAKE_ACTION_BUTTON = i18n.translate(
+  'xpack.securitySolution.alertSummary.flyout.takeActionsAriaLabel',
+  {
+    defaultMessage: 'Take action',
+  }
+);
+export const ADD_TO_CASE_ARIA_LABEL = i18n.translate(
+  'xpack.securitySolution.alertSummary.flyout.attachToCaseAriaLabel',
+  {
+    defaultMessage: 'Attach alert to case',
+  }
+);
+
+/**
+ * Take action button in the panel footer.
+ * This is used in the AI for SOC alert summary page.
+ * The following options are available:
+ * - add to existing case
+ * - add to new case
+ * - apply alert tags
+ */
+export const TakeActionButton: FC = () => {
```

Comment:
any reason you memo'ed the others but not this one?

Reply:
Mistake on my end, will fix as I rebase against latest `main`, thanks for catching this!

Reply:
fixed!